### DISCUSSION TITLE
Upgrading from 0.3.0 to later versions

### DISCUSSION THREAD

**Discussion by bwbuhse** [catppuccin/tmux#339] [Help]
### Is there an existing issue outlining your problem?

- [X] I have searched the existing issues and they do not solve my problem.

### Describe your problem.

The status line is missing the right-hand side modules, as well as process names for tabs.

### Paste your configuration.

```sh
unbind C-b
set -g prefix C-a
bind C-a send-prefix

bind -n C-k clear-history

set-option -g set-titles on
set-option -g set-titles-string "#S / #W"

# set term to tmux instead of gnu screen
# this enables proper italics support
set -ga terminal-overrides ",xterm-256color:Tc"
set -g default-terminal "tmux-256color"

setw -g mode-keys vi
set -g mouse on

set-option -g status-position bottom

set -g @catppuccin_flavor 'mocha'
run ~/.config/tmux/plugins/catppuccin/tmux/catppuccin.tmux

# List of TPM-managaed plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'tmux-plugins/tmux-pain-control'

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run -b '~/.config/tmux/plugins/tpm/tpm'
```

### Attach screenshots.

![Screenshot](https://github.com/user-attachments/assets/fe4066b6-a854-4c2d-b53c-b8e02d36a7cf)

### What tmux version are you seeing the issue on?

tmux 3.4 and tmux 3.5

**Comment by av1155** [comment]
Same here, and my tmux does not want to install cattpuccin via tpm

**Comment by mauromotion** [comment]
Same here, I tried both with and without tpm. With tpm it was like nothing was updated, although it said "updated". Without I could see some differences as in stuff got broken, and some new settings are just doing nothing at all.

**Comment by davidfmatheson** [comment]
I had this problem after updating to 0.3.0, I had to run this task (your location may vary):

```bash
$ ~/.config/tmux/plugins/tpm/bin/clean_plugins
```

After restarting it refreshed the plugins and things were back to normal (thanks to continuum and resurrect plugins).

**Comment by kjnsn** [comment]
Thanks so much for upgrading! I've updated the readme to include some notes regarding the upgrade. v0.4.0 should have fixed the `#T` appearing in the window formats

**Comment by mauromotion** [comment]
Well now after running the cleaning script that @davidfmatheson  mentioned, I can see the update with TPM. Nonetheless it's still not working correctly for me. In the screenshot you can see my tmux/catppuccin config lines and the actual tmux bar. It's on a transparent background but anyway, all the elements in the bar look kinda wrong.

My doubt is: is it the actual current supposed default look of the theme?

![tmux_catppuccin_bugs](https://github.com/user-attachments/assets/bee13c75-0cb6-41ac-b606-207f391c78db)

**Comment by kjnsn** [comment]
Yep that's the default! Although it's odd that the current window style isn't rounded.

Can you explain what you mean by "kinda wrong"? What is the expected behaviour?

**Comment by mauromotion** [comment]
It looks broken? As in, why only the windows are styled? The rest of the bar is just default tmux with a different bg colour. With the previous version of this plugin I had a different default without customising anything (which doesn't mean I liked it, but anyway). 

But if you can confirm this is the "default" look and set up, I can work from here. It's just a weird first experience for a user maybe? It looks inconsistent, half/done half/broken. But maybe it's just me, just my 2 cents. 

I can go on and customise it now (after I figured out why the selected window is squared maybe). Thanks!

**Comment by kjnsn** [comment]
>  (which doesn't mean I liked it, but anyway).

Yeah this was a big reason for changing the defaults.

> But if you can confirm this is the "default" look and set up, I can work from here.

This is the default look and setup currently.

> It's just a weird first experience for a user maybe? It looks inconsistent, half/done half/broken. But maybe it's just me, just my 2 cents.

That's really helpful feedback! I'd be keen to provide some sensible defaults (the current ones could be improved), at least with the initial configuration/examples. Do you have some recommendations for how the defaults could be better?

> I can go on and customise it now (after I figured out why the selected window is squared maybe). Thanks!

Anytime!

**Comment by mauromotion** [comment]
I think a sane default should show most of what is possible to do with the theme. As in, I'd rather start from someone else's customisation than from a blank slate. This is not showing me a good final result that I can possibly achieve with the nice config settings you're giving us. Or maybe, if you'd rather keep a bare bones default for whatever reason, the documentation should be more extensive and with lots of visuals (screenshots) to guide the user. I've just been trying many configs now by throwing the lines in there and reloading. 

Update on the square window that's supposed to be rounded: it gets rounded after I restart tmux completely.

**Comment by kjnsn** [comment]
Done in #335

**Comment by davidfmatheson** [comment]
Version 0.4.0 breaks my setup in a way that I can no longer get back, even after cleaning plugins. I new session looks like this:
<img width="254" alt="image" src="https://github.com/user-attachments/assets/f0f3224b-b818-407d-9819-cb7c364a52d1">

when it should have rounded corners and blue 
![image](https://github.com/user-attachments/assets/ecb90ffd-b115-4776-acda-00f6bb039171)

Can we reopen this issue?

**Comment by mauromotion** [comment]
I honestly switched back to nova theme and I customised that with catpuccin colours. This rework of the theme is much appreciated but I found so many other bugs that it's not worth my time. This new implementation doesn't like tpm for some reason, but that's what I use and prefer. I'd say the rework is going well but it's half way through. 

Will come back in a few days. Thanks.

**Comment by davidfmatheson** [comment]
Pinning to 0.3.0 for now:

```
set -g @plugin 'catppuccin/tmux#v0.3.0'
```

I can get my colors back that way after `clean_plugins`

**Comment by kjnsn** [comment]
> when it should have rounded corners and blue 

Would you mind pasting your config?

As a workaround, you can pin to v0.3.0. I'm not sure if you'll get much value from later versions if what you had worked. Is there a reason you wanted to upgrade?


> Can we reopen this issue?

Done! No worries

**Comment by kjnsn** [comment]
> I honestly switched back to nova theme and I customised that with catpuccin colours. This rework of the theme is much appreciated but I found so many other bugs that it's not worth my time. This new implementation doesn't like tpm for some reason, but that's what I use and prefer. I'd say the rework is going well but it's half way through. 
> 
> 
> 
> Will come back in a few days. Thanks.

Please feel free to file issues and I can take a look. Alternatively you can pin to v0.3.0

**Comment by davidfmatheson** [comment]
> Would you mind pasting your config?
```
# Catppuccin setup
set -g @plugin 'catppuccin/tmux#v0.3.0'
set -g @catppuccin_window_left_separator ""
set -g @catppuccin_window_right_separator " "
set -g @catppuccin_window_middle_separator " █"
set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W"
set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#W#{?window_zoomed_flag,(),}"
set -g @catppuccin_status_modules_right "directory date_time battery"
set -g @catppuccin_status_modules_left "session"
set -g @catppuccin_status_left_separator  " "
set -g @catppuccin_status_right_separator " "
set -g @catppuccin_status_right_separator_inverse "no"
set -g @catppuccin_status_fill "icon"
set -g @catppuccin_status_connect_separator "no"
set -g @catppuccin_directory_text "#{pane_current_path}"
set -g @catppuccin_date_time_text "%H:%M"
set -g @plugin 'tmux-plugins/tmux-battery'
```

Copy/pasting seems to lose the special character config, in my local it looks like this:
<img width="504" alt="image" src="https://github.com/user-attachments/assets/537dce0b-2294-4539-842c-0150e191bcbd">

> As a workaround, you can pin to v0.3.0. I'm not sure if you'll get much value from later versions if what you had worked. Is there a reason you wanted to upgrade?

Generally my setup has `#latest` so refreshing my tmux config upgrades. I also am checking to see if it fixes Issue #327.

**Comment by kjnsn** [comment]
Fair enough. If you wanted to use the latest plugin version you would have to rewrite your config like this:

```
# Catppuccin setup
set -g @plugin 'catppuccin/tmux#latest'
set -g @catppuccin_window_left_separator ""
set -g @catppuccin_window_right_separator " "
set -g @catppuccin_window_middle_separator " █"
set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W"
set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#W#{?window_zoomed_flag,(),}"
set -g @catppuccin_status_left_separator  " "
set -g @catppuccin_status_right_separator " "
set -g @catppuccin_status_right_separator_inverse "no"
set -g @catppuccin_status_fill "icon"
set -g @catppuccin_status_connect_separator "no"
set -g @catppuccin_directory_text "#{pane_current_path}"
set -g @catppuccin_date_time_text "%H:%M"

set -g @plugin 'tmux-plugins/tmux-battery'

set -g status-left "#{@catppuccin_status_session}"
set -g status-right "#{@catppuccin_status_directory}"
set -ag status-right "#{@catppuccin_status_date_time}"
set -ag status-right  "#{@catppuccin_status_battery}"
```

**Comment by davidfmatheson** [comment]
> Fair enough. If you wanted to use the latest plugin version you would have to rewrite your config like this:

Thanks for the config. I tried it and I see the same behavior (non-rounded corners and grey backgrounds).

**Comment by xdannyrobertsx** [comment]
This broke for me yesterday, but was fixed by changing the plugin line to `set -g @plugin 'catppuccin/tmux#tmux-native'`
it looks like it's broken again this morning

here's my config:
```
# globals
set -g mouse on
set -g default-terminal "screen-256color"
set -g base-index 1
set -g pane-base-index 1
set -g word-separators ""
set-option -g prefix C-Space
set-option -g prefix2 C-b

# vim navigation
setw -g mode-keys vi
bind-key v copy-mode
bind-key h select-pane -L
bind-key j select-pane -D
bind-key k select-pane -U
bind-key l select-pane -R
bind-key H previous-window
bind-key L next-window
bind-key J swap-window -t -1\; select-window -t -1
bind-key K swap-window -t +1\; select-window -t +1
bind-key -n S-Left  previous-window
bind-key -n S-Right next-window
bind-key -n S-Up swap-window -t -1\; select-window -t -1
bind-key -n S-Down swap-window -t +1\; select-window -t +1
bind-key '"' split-window -v -c "#{pane_current_path}"
bind-key % split-window -h -c "#{pane_current_path}"

# TMUX Package Manager
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'catppuccin/tmux#tmux-native'
set -g @plugin 'tmux-plugins/tmux-battery'

# Theme
set -g @catppuccin_flavor 'macchiato'
set -g @catppuccin_window_left_separator " "
set -g @catppuccin_window_right_separator ""
set -g @catppuccin_window_middle_separator " "
set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_default_fill "all"
set -g @catppuccin_window_current_fill "all"
set -g @catppuccin_window_default_text "#W"
set -g @catppuccin_window_current_text "#W"
set -g @catppuccin_pane_border_style "fg=#{thm_blue}"
set -g @catppuccin_pane_active_border_style "fg=#{thm_red}"
set -g @catppuccin_status_modules_right "directory battery session"
set -g @catppuccin_session_icon "󱐋"
set -g @catppuccin_status_left_separator  ""
set -g @catppuccin_status_right_separator " "
set -g @catppuccin_status_fill "all"
set -g @catppuccin_status_connect_separator "no"
set -g @catppuccin_directory_color "#{thm_blue}"
set -g @catppuccin_directory_text "#{b:pane_current_path}"
set -g @catppuccin_status_background "default"

# bindings
set-option -g detach-on-destroy off
bind-key S setw synchronize-panes
bind-key . display-popup -E nvim .
bind-key r source-file ~/.config/tmux/tmux.conf \; display "Reloaded!"
bind-key C-l send-keys -R \; send-keys C-l \; clear-history 

# Check if TPM is installed
if "test ! -d $HOME/.config/tmux/plugins/tpm" \
   "run 'git clone https://github.com/tmux-plugins/tpm $HOME/.config/tmux/plugins/tpm && $HOME/.config/tmux/plugins/tpm/bin/install_plugins'"

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '$HOME/.config/tmux/plugins/tpm/tpm'
```

Editing my tpm line to set -g @plugin 'catppuccin/tmux#v0.3.0' and running `./clean_plugins` like mentioned above works

**Comment by kjnsn** [comment]
If you'd like to keep on using v0.3.0 you are more than welcome to. It's tagged for a reason :)

Otherwise you will have to migrate your config to be compatible. Please see #291 for more details

**Comment by Rasmus-Bertell** [comment]
I have problems with the window separators and pomodoro_plus, everything else seems to work after migration. The options `@catppuccin_window_left_separator` and `@catppuccin_window_right_separator` seem to have no impact. Tested also with all the example configurations and none of them looked like their pictures. Pomodoro status doesn't change and the colors are weird, the plugin `olimorris/tmux-pomodoro-plus` works fine I get notifications of the status changes but they are not reflected in tmux status bar.

I'm currently running tmux inside WSL but I'll report how it works on native linux once I get around to updating my other computer.

![image](https://github.com/user-attachments/assets/7a698c8e-7a6a-422d-bbff-c9c5721651e6)
![image](https://github.com/user-attachments/assets/5a0e078a-61d5-43f7-8d9c-c75652d7dca2)

My config:

```tmux
#...
set-option -g @plugin 'catppuccin/tmux'
set-option -g @plugin 'olimorris/tmux-pomodoro-plus'
#...

# Catppuccin options
set-option -g @catppuccin_flavor 'latte'

set-option -g @catppuccin_window_left_separator ""
set-option -g @catppuccin_window_right_separator " "

set-option -g @catppuccin_window_default_text "#W"
set-option -g @catppuccin_window_current_text "#W"

set-option -g @catppuccin_status_modules_right "pomodoro_plus application directory user host session"
set-option -g @catppuccin_status_left_separator  ""
set-option -g @catppuccin_status_right_separator ""
set-option -g @catppuccin_status_connect_separator "no"

set-option -g @catppuccin_directory_text "#{pane_current_path}"

set-option -g  status-right "#{E:@catppuccin_status_pomodoro_plus}"
set-option -ag status-right " #{E:@catppuccin_status_kube}"
set-option -ag status-right " #{E:@catppuccin_status_application}"
set-option -ag status-right " #{E:@catppuccin_status_directory}"
set-option -ag status-right " #{E:@catppuccin_status_user}"
set-option -ag status-right " #{E:@catppuccin_status_host}"
set-option -ag status-right " #{E:@catppuccin_status_session}"

# Pomodoro Plus options
set-option -g @pomodoro_on              '󰐊 '
set-option -g @pomodoro_complete        '󰶞 '
set-option -g @pomodoro_pause           '󰏤 '

set-option -g @pomodoro_prompt_break    'break?'
set-option -g @pomodoro_prompt_pomodoro 'start?'

set-option -g @pomodoro_long_break_mins 15

set-option -g @pomodoro_notifications 'on'

#...

run '~/.config/tmux/plugins/tpm/tpm'
```

**Comment by Rasmus-Bertell** [comment]
Figured out why window styles weren't working. I had to define `@catppuccin_window_status_style` as stated in https://github.com/catppuccin/tmux/tree/main?tab=readme-ov-file#window. Should also be reflected in the example configs because they don't work without it and in the migration guide.

Also figured out how to get rid of the session title next to the windows by defining 

```tmux
set-option -g status-left ""
```

This just leaves me with the pomodoro plus status widget which doesn't work at all but I can open a separate issue for that.

**Comment by kjnsn** [comment]
If you're happy to create a separate issue for pomodoro that would be great!

**Comment by davidfmatheson** [comment]
Ok, I have some of my configuration rebuilt with this new version. For reference, I'm trying to get back to this:
<img width="801" alt="image" src="https://github.com/user-attachments/assets/d72b4810-71c3-4320-8b55-41ceba1dcf1a">

and right now the configuration that achieves this pins to v0.3.0 (for reference it's over [here](https://github.com/davidfmatheson/.dotfiles/blob/main/tmux/.config/tmux/tmux.conf.local#L432)). My new setup is just missing the blue on non-active windows, the darker grey backgrounds, and the battery icon:
<img width="803" alt="image" src="https://github.com/user-attachments/assets/5fbee3c3-d703-4d8a-a00d-47dfb2030c82">

So far my rewrite looks like this:
```
# Catppuccin setup
set -g @plugin "catppuccin/tmux#latest"

set -g @catppuccin_window_status_style "custom"
set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_left_separator ""
set -g @catppuccin_window_right_separator " "
set -g @catppuccin_window_middle_separator " █"

set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W"

set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#W#{?window_zoomed_flag,(),}"

set -g @catppuccin_status_left_separator  " "
set -g @catppuccin_status_right_separator ""

set -g @catppuccin_status_right_separator_inverse "no"
set -g @catppuccin_status_connect_separator "no"
set -g @catppuccin_status_fill "icon"
#
set -g status-right "#{E:@catppuccin_status_directory}#{E:@catppuccin_status_date_time}#{E:@catppuccin_status_battery}"
set -g status-left "#{E:@catppuccin_status_session}"

set -g @catppuccin_directory_text "#{pane_current_path}"
set -g @catppuccin_date_time_text "%H:%M"

set -g @plugin "tmux-plugins/tmux-battery"
```

**Reply by kjnsn** [reply]
Out of curiosity, why do you want to upgrade?

**Reply by mauromotion** [reply]
> Out of curiosity, why do you want to upgrade?

This is not directed to me but I feel compelled to reply because you keep wondering this even in the discord server: 

why are you developing this any further at all? If you answer this question you might understand why people want to upgrade. 

Why are you so surprised?

People like to try new things, and improved things, that might be better than before. You are giving us easier ways to configure the theme with this update so people like it and want to upgrade. Of course if it doesn't work properly people will either move somewhere else, keep the previous version, try to help.

Also, coming from the previous version's documentation we all had the recommended tag @latest, and so the update kinda showed itself up for a lot of people.

**Reply by sgoudham** [reply]
My two cents here is that we should get release please to automatically update the version in the README when a release is cut. 

Apologies for missing that in my review for introducing release please, it would have most likely avoided people pinning latest by default.

I'm almost tempted to get rid of that tag entirely. 

With regards to why people want to upgrade, I think it's to be expected since we're still pushing out new updates but we should be a bit more clear in our communication that existing users don't necessarily need to upgrade if what they have works already. Of course, we should help them if they're up for it and that's where a nice migration guide would come in and what these discussions are for.

**Reply by davidfmatheson** [reply]
> Out of curiosity, why do you want to upgrade?

I was hoping for a fix for Issue #327, but before I can test that I have to get back to my previous starting state, then try resurrecting.

**Comment by bwbuhse** [comment]
Is there a config to just mimic the previous default behavior/colors/modules? I supposed I could always downgrade (like has been suggested), but I'm also just wondering. I mostly just like catppuccin for its colors and don't like configuring tmux too much.

**Reply by kjnsn** [reply]
As the plugin develops, default colors and behaviors are tweaked to be a bit "better", i.e. being more compatible with recent tmux updates, or to improve color contrast. I've been following a policy not to change the overall colors too much, but I did update the window colors to improve readability and contrast.

I've also been attempting to make the plugin easier to configure, and to have more pleasing defaults, which will hopefully reduce the amount of customisation required.

**Reply by davidfmatheson** [reply]
This got me pretty close to my previous configuration:
```
set -g @plugin "catppuccin/tmux#latest"

set -g @catppuccin_window_status_style "custom"
set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_left_separator ""
set -g @catppuccin_window_right_separator ""
set -g @catppuccin_window_middle_separator " █"
set -g @catppuccin_window_default_background "#{@thm_blue}"
set -g @catppuccin_window_default_color "#{@thm_surface_0}"
set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W"
set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#W#{?window_zoomed_flag,( ),}"
set -g @catppuccin_window_current_color "#{@thm_surface_1}"
set -g @catppuccin_window_current_background "#{@thm_peach}"

set -g @catppuccin_status_left_separator  " "
set -g @catppuccin_status_right_separator " "
set -g @catppuccin_status_right_separator_inverse "no"
set -g @catppuccin_status_connect_separator "no"
set -g @catppuccin_status_fill "icon"

set -g status-right "#{E:@catppuccin_status_directory}#{E:@catppuccin_status_date_time}#{E:@catppuccin_status_battery}"
set -g status-left "#{E:@catppuccin_status_session}"

set -g @catppuccin_directory_text "#{pane_current_path}"
set -g @catppuccin_date_time_text "%H:%M"

set -g @plugin "tmux-plugins/tmux-battery"

# turn off visual notification because it alters colors
set -g visual-bell off
setw -g monitor-activity off
```

<img width="1049" alt="image" src="https://github.com/user-attachments/assets/216a9805-2fed-4394-8f36-dd3a2a2d2446">

**Reply by davidfmatheson** [reply]
Any recommendations to get the battery icon & percentage to show up on the right-most status module?

**Reply by kjnsn** [reply]
Yep: https://github.com/catppuccin/tmux/discussions/317#discussioncomment-10856713

Please note that most non-basic status line modules will be removed in a later release with a preference for config like above.

**Comment by davidfmatheson** [comment]
If it helps, [here is a PR](https://github.com/davidfmatheson/.dotfiles/pull/1/files) I have in my `.dotfiles` repo that shows the changes I made between 0.3.0 and latest to get things working. I am still working on the battery % on the far right (seems to be Issue #342) and the session icon on the far left, but the rest is pretty identical to what I had.

**Reply by kjnsn** [reply]
Nice! We'd be super appreciative of any improvements to the docs based on your experience :)
